### DISCUSSION TITLE
How to prevent 'idle mode' if capacity level is higher than 'pause_capacity'

### DISCUSSION THREAD

**Discussion by hanni2301** [VR-25/acc#150] [Q&A] [answered]
Hi, first of all thank you for your great effort you putting in this project!

I'm getting use to it on all my smartphones and also was finding the right charging switches for each device.

Actually I'm fiddling with my Pixel 5.

If I plug the charger in at a charging (capacity) level of the battery which is lower then the desired  'resume_capacity' level in the config, the device will charge up fine until its  'pause_capacity' level and will enter then idle mode (if the charging switch does support this) or will stop charging until the capacity level falls below its threshold.
So far so good.

If I now change the charging profile (with AccA for example) and the capacity level is higher than the configured capacity level in the config the device will enter directly into idle mode if I plug in the charger.

Because I like to just let the battery drain itself to the capacity level I configured in the config and prevent staying at a higher percentage (idle mode), I'm looking for a way to get the best out of both worlds. 

One example.
Battery is at 75 % at 8 pm.
Set capacity=(20 101 49 50 true false)
Charging switch supports idle mode (or does not, I don't know what will work)
Plugging in the charger at 9 pm
**Battery should drain itself while it's night to get some stress of it - so idle mode should not kick in!**
At 6 am I let the charging profile switch to: capacity=(20 101 74 75 true false)
Now I should find my decive in idle mode at 75 % at any hour later than like 7 am.

How do I get these highlighted behaviour?
I know that o could just use a charging switch which doesn't support idle mode, but then I will get this  infinite loop of charging cycles.

Thanks in advance!

**Comment by VR-25** [comment, marked as answer]
Hi,
The latest dev build has this feature.
One does not have to do anything to enable it. ACC will always avoid idle mode above `pause_capacity`, if _possible_*.

\* Some devices only have _idle mode_ switches. The Motorola Moto G4 Play is an example.

[acc_v2022.5.25-dev_202205250_0935.zip](https://github.com/VR-25/acc/files/8769549/acc_v2022.5.25-dev_202205250_0935.zip)
https://t.me/+jN081M2459M5Mjhk (a Telegram group for testing dev builds).
